scene/src/utils/URN-utils.ts

```diff
@@ -0,0 +1,46 @@
+import type { URN, URNWithoutTokenId } from './definitions'
+export const BASE_MALE_URN: URNWithoutTokenId =
+  'urn:decentraland:off-chain:base-avatars:BaseMale' as URNWithoutTokenId
+
+const urnWithoutTokenIdMemo: Record<URN, URNWithoutTokenId> = {} // TODO consider using Map if possible for performance improvement because long keys
+export const urnWithTokenIdMemo: Record<URNWithoutTokenId, URN> = {}
+
+export function getWearablesWithTokenId(wearables: URNWithoutTokenId[]): URN[] {
+  return wearables.map((wearableURN) => urnWithTokenIdMemo[wearableURN])
+}
+
+export function getURNWithoutTokenId(
+  urn: URN | null | URNWithoutTokenId,
+  avoidCache: boolean = false
+): URNWithoutTokenId | null {
+  if (urn === null) return null
+
+  // Check if the value is already cached and avoid cache flag is false
+  if (urnWithoutTokenIdMemo[urn as URN] !== undefined && !avoidCache) {
+    return urnWithoutTokenIdMemo[urn as URN]
+  }
+
+  // Check if it's an off-chain URN or has less than 6 parts
+  if (urn.includes(':off-chain:') || urn.split(':').length < 6) {
```

Comment:
on checks like this, is great to put functions with the name of the action that you are going to do... so the code explains by itself... like

```typescript
function isOffChainOrIncomplete(urn: string): boolean {
  return urn.includes(':off-chain:') || urn.split(':').length < 6
}
```

and the result looks more clear

```typescript
if (isOffchainOrIncomplete(urn)) { /* ... */ }